java-7/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM openjdk:7-jdk-alpine
+
+RUN wget -O /dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.1/dumb-init_1.2.1_amd64
+RUN chmod +x /dumb-init
+
+ENV LC_ALL="no_NB.UTF-8"
+ENV LANG="no_NB.UTF-8"
+ENV TZ="Europe/Oslo"
+
+# Please see https://blogs.oracle.com/java-platform-group/java-se-support-for-docker-cpu-and-memory-limits
+ENV DEFAULT_JAVA_OPTS="-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -XX:ParallelGCThreads -XX:CICompilerCount"
```

Comment:
Kan du undersøke om `-XX:+UseCGroupMemoryLimitForHeap`  er støttet med versjonen `openjdk:7-jdk-alpine` kommer med?
På versjonen jeg har installert (1.7.0_79) så avslutter den med:
`Unrecognized VM option 'UseCGroupMemoryLimitForHeap'`

Det flagget kom først i 1.8.0_131 men det er mulig at OpenJDK har patchet det inn i senere versjoner av JDK7 (jeg vet ikke)

Hvis det flagget ikke fungerer med flagg så anbefaler jeg å tilpasse script (slik Giao skrev på slack) for å finne ut av spesielt minne, fra https://github.com/fabric8io-images/java/tree/master/images/jboss/openjdk7/jdk .
Da vi testet dette ut i starten da NAIS var nytt så gikk apper i OOMKilled crashloops pga Java i Docker trodde at fysisk minne tilgjengelig var 100% av noden den kjørte på i Kubernetes..

Forslag:

* Sjekk at disse flaggene fungerer med denne JDK7 før vi merger

Reply:
Ser ut som at det bare er UseCGroupMemoryLimitForHeap som ikke er støttet. Men vi trenger kanskje ikke UnlockExperimentalVMOptions lenger når vi tar den bort? 

I følge https://blogs.oracle.com/java-platform-group/java-se-support-for-docker-cpu-and-memory-limits ser det ut som at vi kanskje må spesifisere minnet med -Xmx. Må nesten se litt på hvordan jeg kan få til det.

Reply:
Du vil måtte beregne cpu cores itillegg til xmx verdi. Da flagg som XX:ParallelGCThreads, krever en verdi. XX:ParallelGCThreads=4 f.eks.

Eksempel her (xmx):
https://github.com/finn-no/tomcat7-docker/blob/master/deploy-and-run.sh

og her(xms og cpu_cores):
https://github.com/fabric8io-images/java/blob/master/images/alpine/openjdk7/jdk/run-java.sh